partner_relations/model/res_partner.py

```diff
@@ -298,12 +298,11 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None,
             cr, uid, args + date_args + active_args, offset=offset,
             limit=limit, order=order, context=context, count=count)
 
-    def read(
-            self, cr, uid, ids, fields=None, context=None,
-            load='_classic_read'):
-        return super(ResPartner, self).read(
-            cr, uid, ids, fields=fields,
-            context=self._update_context(context, ids), load=load)
+    @api.multi
+    def read(self, fields=None, load='_classic_read'):
+        self.env.context = self._update_context(
+            self.env.context, [r.id for r in self])
+        return super(ResPartner, self).read(fields=fields, load=load)
```

Comment:
@eLBati have a look at this gist: https://gist.github.com/hbrunn/174be7844b409666a281
There you will see how values in your context don't match values in the cache. This is asking for trouble.